stubs/docutils/docutils/nodes.pyi

```diff
@@ -113,40 +135,234 @@ class Element(Node):
+class line_block(General, Element): ...
+
+class line(Part, TextElement):
+    indent: str | None
+
+class block_quote(General, Element): ...
+class attribution(Part, TextElement): ...
+class attention(Admonition, Element): ...
+class caution(Admonition, Element): ...
+class danger(Admonition, Element): ...
+class error(Admonition, Element): ...
+class important(Admonition, Element): ...
+class note(Admonition, Element): ...
+class tip(Admonition, Element): ...
+class hint(Admonition, Element): ...
+class warning(Admonition, Element): ...
+class admonition(Admonition, Element): ...
+class comment(Special, Invisible, FixedTextElement): ...
+class substitution_definition(Special, Invisible, TextElement): ...
+class target(Special, Invisible, Inline, TextElement, Targetable): ...
+class footnote(General, BackLinkable, Element, Labeled, Targetable): ...
+class citation(General, BackLinkable, Element, Labeled, Targetable): ...
+class label(Part, TextElement): ...
+class figure(General, Element): ...
+class caption(Part, TextElement): ...
+class legend(Part, Element): ...
+class table(General, Element): ...
+class tgroup(Part, Element): ...
+class colspec(Part, Element): ...
+class thead(Part, Element): ...
+class tbody(Part, Element): ...
+class row(Part, Element): ...
+class entry(Part, Element): ...
+
+class system_message(Special, BackLinkable, PreBibliographic, Element):
+    def __init__(self, message: str | None = None, *children: Node, **attributes) -> None: ...
+
+class pending(Special, Invisible, Element):
+    transform: Transform
+    details: Mapping[str, Any]
+    def __init__(
+        self, transform: Transform, details: Mapping[str, Any] | None = None, rawsource: str = "", *children: Node, **attributes
+    ) -> None: ...
+
+class raw(Special, Inline, PreBibliographic, FixedTextElement): ...
+
+# Inline Elements
+
+class emphasis(Inline, TextElement): ...
+class strong(Inline, TextElement): ...
+class literal(Inline, TextElement): ...
+class reference(General, Inline, Referential, TextElement): ...
+class footnote_reference(Inline, Referential, TextElement): ...
+class citation_reference(Inline, Referential, TextElement): ...
+class substitution_reference(Inline, TextElement): ...
+class title_reference(Inline, TextElement): ...
+class abbreviation(Inline, TextElement): ...
+class acronym(Inline, TextElement): ...
+class superscript(Inline, TextElement): ...
+class subscript(Inline, TextElement): ...
+class math(Inline, TextElement): ...
+class image(General, Inline, Element): ...
+class inline(Inline, TextElement): ...
+class problematic(Inline, TextElement): ...
+class generated(Inline, TextElement): ...
+
+# Auxiliary Classes, Functions, and Data
+
+node_class_names: list[str]
+
+# necessary to disambiguate type and field in NodeVisitor
+_Document = document
 
 class NodeVisitor:
-    def __init__(self, document: document): ...
+    optional: ClassVar[tuple[str, ...]]
+    document: _Document
```

Comment:
IDK if there’s a better way to do this

Reply:
I don't think it's needed for the field. But AFAIK there's no better way to deal with this for the parameter name (line below). See #11113